Add French translation

joggee-fr opened the pull request:
This is my proposal to add a French translation to FreeOTP.
I sometimes had to not stick to the English base sentences to make it clearer in French.

BTW, is there  difference between "issuer" and "provider"? I decided to stay with "fournisseur" in French which is closer to "provider". A possible translation for "issuer" could be "émetteur".

justin-stephenson commented:
> This is my proposal to add a French translation to FreeOTP. I sometimes had to not stick to the English base sentences to make it clearer in French.

Thank you for the PR.
> 
> BTW, is there difference between "issuer" and "provider"? I decided to stay with "fournisseur" in French which is closer to "provider". A possible translation for "issuer" could be "émetteur".

`Issuer` refers to this specific OTP URI field https://github.com/google/google-authenticator/wiki/Key-Uri-Format#issuer
which essentially means the same thing as the OTP provider.    _The issuer parameter is a string value indicating the provider or service this account is associated with_

If no more changes are needed I will go ahead and merge this, just let me know.

joggee-fr commented:
@justin-stephenson, thanks for the detail.
Let me triple-check my proposal in this case. Would be great if another French-speaking contributor could give it a look. Pinging @futal from PR #174 for example.

Patriccollu commented on mobile/src/main/res/values-fr/strings.xml:
```diff
@@ -0,0 +1,125 @@
+    <string name="restore_title">Restaurer</string>
+    <string name="backup_restore_title">Sauvegarde et restauration</string>
+    <string name="keystore_migration_explanation">Les jetons FreeOTP ont désormais migré vers le magasin de clés Anroid\u00A0! Il en résulte une sécurité accrue, mais les clés ne sont alors plus exportables. \n\nLes fonctionnalités de sauvegarde et restauration sont désormais ajoutées afin d\'assurer la récupération des jetons. Vous serez invité à créer un mot de passe maitre pour initialiser les ...
+    <string name="manual_backup_explanation">Parallèlement, un fichier de sauvegarde chiffré peut être exporté sur le stockage externe depuis l\'applicationan. La barre d\'outils supérieure contient désormais les entrées nommées \'Sauvegarde\' et \'Restauration\' afin d\'effectuer cette sauvegarde manuelle.</string>
```
... depuis l\'application.

Patriccollu commented:
Hello

I've just translated the file in Corsican but I have good knowledges in French ;-)

In my opinion, there is no problem to translate issuer by « Fournisseur ».

I noticed a couple of typos on lines 114 and 118 and I added a comment on your file.

Best regards,
Patriccollu.

joggee-fr commented:
> I've just translated the file in Corsican but I have good knowledges in French ;-)

Thanks for your review @Patriccollu.
Rebased, triple-checked and fixed the typos.

@justin-stephenson, my proposal is now ready to be merged if you are OK with it.

futal commented:
Hi. Sorry for the late reply. I confirmed the changes suggested by @Patriccollu. Here are some more typos I found in mobile/src/main/res/values-fr/strings.xml:
* Missing accent (mandatory as it is another word without the accent) in line 65 "Cela peut être du à un changement de paramètres du verrouillage de l\'écran." -> "Cela peut être dû à un changement de paramètres du verrouillage de l\'écran."
* Missing letter in line 117 "Les jetons FreeOTP ont désormais migré vers le magasin de clés Anroid\u00A0!" -> "Les jetons FreeOTP ont désormais migré vers le magasin de clés Android\u00A0!"
* Translations on lines 120 and 121 are not good but the expected meaning is not clear enough for me to come with better translations.

joggee-fr commented:
Thanks @futal.
- Added missing accent line 65 (_dû_) ;
- Added missing __d__ line 117 (__Android__).

As I understant it, lines 120 and 121 are content description for images (just like __alt__ property for HTML tag __img__ for example). I don't think they are really displayed but could be nice for accessibility, screen reader and stuff like this.

futal commented:
Then lines 120 and 121 are fine for alt properties (it is not clear out of context as there are no uppercase after "Image" but they become clear as alt tags which are usually pronounced with a different voice and the upper case does not matter there).

joggee-fr commented:
Thanks again for yours reviews @Patriccollu and @futal.
